### ISSUE TITLE
Parse git config from yadm config

### ISSUE THREAD

**Issue by xerus2000** [TheLocehiliosan/yadm#251]
### Is your feature request related to a problem? Please describe.

I would like to define aliases and other git config specifically for yadm.

### Describe the solution you'd like

Add `$XDG_CONFIG_HOME/yadm/config` as git config file when using yadm.

### Describe alternatives you've considered

Put the alias into my global git config - works, but pollutes it unneccessarily. And in case I want different configuration in some keys, this falls apart.

Another alternative is a git config file local to the yadm repo, but that wouldn't sync, which is a huge annoyance.

**Comment by TheLocehiliosan** [comment]
This is an interesting idea. All of the supported configurations are namespaces into a `[yadm]` section so they can easily overlap with standard Git configuration sections.

I'm not sure there is a way to use both the standard Git global config (`$XDG_CONFIG_HOME/git/config` & `~/.gitconfig`) and also another custom specified config. I believe using `$GIT_CONFIG` merely overrides those paths.

Most users want aliases and configurations they normally use for Git to be available for yadm. That would require users to maintain common configurations in two files (or at least duplicate an `[include]` section in both).

However, I just tried to pass in a config `-c include.path=$HOME/.config/yadm/config`, and it looks like that works. This preliminary test shows that the include is done last, and also results in the yadm config being able to override other settings.

More testing is necessary, but this looks like it might be possible to do by always adding a `-c include.path` option to every Git command.

**Comment by TheLocehiliosan** [comment]
@xerus2000 You can apply this [patch](https://gist.githubusercontent.com/TheLocehiliosan/92a1e8a17c1b2d53c5b59faf118e6ad3/raw/3410bac8f3010cf03996830d0041fe13de5c08c8/0001-Test-inclusion-of-yadm-config-for-standard-Git-comma.patch) to yadm if you want to try this out now. More testing and scrutiny is necessary before bringing into the codebase, but if you want this ability right now, try it out.

**Comment by xerus2000** [comment]
> Most users want aliases and configurations they normally use for Git to be available for yadm. That would require users to maintain common configurations in two files (or at least duplicate an `[include]` section in both).

Yes, I would like the config to be used in addition to (potentially overriding) the values in the global git config.

`include.path` sounds like the correct solution. I will try it soon, thanks :)

**Comment by erijo** [comment]
This use-case can be solved by using the standard git config and includeIf. Put something like this in ~/.gitconfig:
```
[includeIf "gitdir:~/.config/yadm/repo.git"]
	path = .gitconfig.yadm
```

And then put the aliases you want in ~/.gitconfig.yadm:
```
[alias]
        stu = status
```

**Comment by jgogstad** [comment]
> This use-case can be solved by using the standard git config and includeIf. Put something like this in ~/.gitconfig:
> 
> ```
> [includeIf "gitdir:~/.config/yadm/repo.git"]
> 	path = .gitconfig.yadm
> ```

Repo now lives in `~/.local`:

```
❯ tail -n2 .gitconfig
[includeIf "gitdir:~/.local/share/yadm/repo.git"]
        path = ~/.gitconfig.yadm
```
